enterprise/admin.py

```diff
@@ -36,13 +37,54 @@ class EnterpriseCustomerBrandingConfigurationInline(admin.StackedInline):
     can_delete = False
 
 
+class ModelChoiceIntegerField(forms.ModelChoiceField):
+    """
+    A custom integer field for admin form, this field treats an integer field as a foreign key.
+    """
+
+    def to_python(self, value):
+        """
+        Return an integer instead of a model instance.
+
+        Since we are emulating Foreign key from an Integer Field we need to return integer instead of fetching the
+        model instance.
+        """
+        try:
+            return int(value)
+        except (ValueError, TypeError):
+            return None
+
+
+class EnterpriseCustomerForm(forms.ModelForm):
+    """
+    A custom model form to emulate a foreign key on an integer field.
+
+    A model form that converts an integer field to a foreign key if a particular
+    model is present in the system.
+    """
+
+    try:
+        # We give a choice field to user only if SAMLProviderConfig is present, otherwise we show
+        # an integer field, Since, we will be adding a custom enterprise admin so we will also be removing
+        # this dependency.
+        from third_party_auth.models import SAMLProviderConfig
+        identity_provider = ModelChoiceIntegerField(SAMLProviderConfig.objects.all(), required=False)
```

Comment:
Should not we ONLY fetch those `SAMLProviderConfig` that are `enabled` in system ?